cmake/environmentchecks.cmake

```diff
@@ -14,21 +14,54 @@
 
 find_package(Python ${PREFERRED_PYTHON_VERSION} COMPONENTS Interpreter Development)
 
-find_program(VALGRIND_EXE NAMES ${PREFERRED_VALGRIND_NAMES} valgrind)
-find_program(GCOV_EXE NAMES ${PREFERRED_GCOV_NAMES} gcov)
+find_program(CLANG_TIDY_EXE NAMES ${PREFERRED_CLANG_TIDY_NAMES} clang-tidy)
 find_program(FIND_EXE NAMES ${PREFERRED_FIND_NAMES} find)
+find_program(GCOV_EXE NAMES ${PREFERRED_GCOV_NAMES} gcov)
+find_program(VALGRIND_EXE NAMES ${PREFERRED_VALGRIND_NAMES} valgrind)
+
 find_package(Doxygen)
 find_package(Sphinx)
 find_package(SWIG 3)
 
-if(VALGRIND_EXE AND Python_Interpreter_FOUND)
-  set(VALGRIND_TESTING_AVAILABLE TRUE CACHE BOOL "Executables required to run valgrind testing are available.")
-endif()
-if(GCOV_EXE AND FIND_EXE AND Python_Interpreter_FOUND)
+set(HAVE_LIBXML2_CONFIG FALSE)
+if (MSVC)
+  # If we want to use config packages on Windows with Visual Studio,
+  # we need to have two find_package calls and explicitly state that
+  # we wish to use Config mode in the first call.  Finding LibXml2 in config mode
+  # is the preferred method so we will try this first quietly.
+  #
+  # This does change how we get information about include paths and such so we
+  # need to track how we found LibXml2.
+  find_package(LibXml2 CONFIG QUIET)
+  if (LibXml2_FOUND)
+    set(HAVE_LIBXML2_CONFIG TRUE)
+  else ()
+    find_package(LibXml2 REQUIRED)
```

Comment:
Can we add in here:

`unset(LibXml2_DIR CACHE)`

Reply:
There are several calls to `find_package(LibXml2...)`. So, are you meaning that we should that `unset()` call between `set(HAVE_LIBXML2_CONFIG FALSE)`and `if(MSVC)`?